### PR TITLE
Delivery return feature

### PR DESCRIPTION
**Pull request by gotoinc**

### CODE REVIEW

File: app/controllers/deliveries/pending_controller.rb [hunk 1]
```diff
@@ -3,7 +3,9 @@ class Deliveries::PendingController < ApplicationController
   before_action :location
 
   def index
-    @deliveries = Delivery.pending_and_in_the_future.where(pickup_address: current_user.addresses).or(Delivery.pending_and_in_the_future.where(assigned_rider_id: current_user.id, rider_id: nil))
+    @deliveries = Delivery.pending_and_in_the_future.where(pickup_address: current_user.addresses).
+      or(Delivery.pending_and_in_the_future.where(delivery_address: current_user.addresses.map(&:one_line), rider_id: nil)).
```
Comment (iferunsewe) [inline app/controllers/deliveries/pending_controller.rb:7]: I'm not sure this is a great way to do this because delivery addresses might not be the same as pickup addresses. I definitely have to make delivery addresses structured data and its my mistake that they are not. However, until then I think we have to do differently because delivery addresses don't always == pick up addresses

File: app/views/deliveries/form/_address_without_pickup_address_options.html.erb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <div>
-  <div id="delivery_pickup_address" class="d-none"><%= pickup_address.one_line %></div>
-  <%= form.hidden_field :pickup_address_id, value: pickup_address.id %>
-</div>
+  <%= form.label :pickup_address_id, "Pickup address *", style: "display: block", class: 'form-label' %>
```
Comment (iferunsewe) [inline app/views/deliveries/form/_address_without_pickup_address_options.html.erb:2]: Why were these changes needed?

File: app/views/deliveries/form/_rider.html.erb [hunk 3]
```diff
@@ -2,7 +2,8 @@
   <div class="col-9">
     <div class="form-group mb-3">
       <div class="form-floating">
-        <%= form.select :rider_id, User.where(role: ["admin", "rider"]).map { |u| [u.full_name, u.id] }, {prompt: "-"}, {class: "form-control form-select"} %>
+        <%= form.select :rider_id, options_for_select([["-", nil]] + User.where(role: ["admin", "rider"]).map { |u| [u.full_name, u.id] },
```
Comment (iferunsewe) [inline app/views/deliveries/form/_rider.html.erb:5]: Instead of "-", please can you use "Please select a reason?"

Comment (gotoinc) [reply app/views/deliveries/form/_rider.html.erb:5]: Will be fixed

File: app/controllers/deliveries/pending_controller.rb (hunk 1 above)
Comment (gotoinc) [reply app/controllers/deliveries/pending_controller.rb:7]: My opinion is a little over our task. Because changing the string field to the Address object field should be covered tests as manual and auto. It takes a bit more time to fix all functionality in the whole app not just in our return feature. I would like to limit myself to the solution we have.
We would gladly work on it as a part of the main scope.

File: app/views/deliveries/form/_address_without_pickup_address_options.html.erb (hunk 2 above)
Comment (gotoinc) [reply app/views/deliveries/form/_address_without_pickup_address_options.html.erb:2]: At first, I thought this template was not used, so I decided to rework it for the return delivery template. But no, I'm thinking of creating 1 more partial template for this case.

File: db/schema.rb [hunk 4]
```diff
@@ -151,7 +162,7 @@
...
     t.integer "max_travel_time_driving", default: 30
     t.integer "default_vehicle_type", default: 0
-    t.jsonb "week_days_options", default: {"1"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "2"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "3"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "4"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "5"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "6"=>{"weekend"=>true, "working_hours"=>{"open"=>8, "close"=>17}}, "7"=>{"weekend"=>true, "working_hours"=>{"open"=>8, "close"=>17}}}, null: false
+    t.jsonb "week_days_options", default: {"1"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "2"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "3"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "4"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "5"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "6"=>{"weekend"=>false, "working_hours"=>{"open"=>8, "close"=>17}}, "7"=>{"weekend"=>true, "working_hours"=>{"open"=>8, "close"=>17}}}, null: false
```
[the remark this answers is not in the document]
Comment (gotoinc) [reply db/schema.rb:165]: It depends on the DEFAULT_WEEK_DAYS_OPTIONS constant in the team.rb file 
I haven't changed anything here. Therefore, I think you made changes to this constant and did not add a new scheme to the commit.

File: app/views/deliveries/form/_address_without_pickup_address_options.html.erb (hunk 2 above)
Comment (gotoinc) [reply app/views/deliveries/form/_address_without_pickup_address_options.html.erb:2]: Because the return feature means the we have swapped addresses. We don't need any options here. And I modified the file *_without_options*

File: app/controllers/deliveries/pending_controller.rb (hunk 1 above)
Comment (iferunsewe) [reply app/controllers/deliveries/pending_controller.rb:7]: But I'm not asking you to fix the whole functionality but just this query. I'm fine to try it out but it may need to be changed

File: app/views/deliveries/form/_address_without_pickup_address_options.html.erb (hunk 2 above)
Comment (iferunsewe) [reply app/views/deliveries/form/_address_without_pickup_address_options.html.erb:2]: But I don't think this needs to change as you can just use the partial "_address_with_pickup_address_options.html"

File: db/schema.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (iferunsewe) [reply db/schema.rb:165]: Yeah, I changed it in the model but not in the migration this should remain the same. Can you remove this change?

File: app/views/deliveries/_delivery.html.erb [hunk 5]
```diff
@@ -64,6 +64,13 @@
             <%= raw t("delivery.show.payment-failed-message", payment_link: delivery.payment_link) %>
           <% else %>
             <%= t("delivery.show.#{@delivery.status}-message") %>
+              <% if @delivery.returned? %>
+                Check the returned delivery
```
Comment (iferunsewe) [inline app/views/deliveries/_delivery.html.erb:68]: Can this go in the en.yml?

Comment (gotoinc) [reply app/views/deliveries/_delivery.html.erb:68]: Of course.

**Review by iferunsewe — APPROVED** [review]
